### ISSUE TITLE
Extra type checking in train_test_split()

### ISSUE THREAD

**Issue by EvanBrown4** [EvanBrown4/cmor-438-fall#7]
### Feature Summary
Add accurate dimensionality checking for X and y. X should be 2d and y should be 1d.

### Motivation
X needs to have dimensions of (n_samples, n_features). y must have one label per sample in X.

### Implementation Ideas (Optional)
Add helpers (as they will likely be used elsewhere as well) within src/supervised_learning/utilities/preprocess.py that check the dimensionality of an array.

---
Thank you for helping improve this repo!~
